api/agent/agent.go

```diff
@@ -965,10 +970,12 @@ func (a *agent) runHotReq(ctx context.Context, call *call, state ContainerState,
 					return false
 				}
 				isFrozen = true
+				state.UpdateState(ctx, ContainerStatePaused, call.slots)
 			}
 			continue
 		case <-evictor.C:
 			logger.Debug("attempting hot function eject")
+			isEvictEvent = true
```

Comment:
i guess it's a timing thing but uh, `statsContainerEvicted(ctx)` here  will run really fast (it's an atomic add) and then we don't have to drag that boolean variable around.

Reply:
Extra bool is because the eviction event may arrive here, but we have to also check if slot was grabbed a few lines below.